app/Commands/PullCommand.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Commands;
+
+use App\Services\MiseService;
+use Illuminate\Support\Facades\Process;
+use Illuminate\Support\Facades\Storage;
+use LaravelZero\Framework\Commands\Command;
+
+use function Laravel\Prompts\confirm;
+use function Laravel\Prompts\error;
+use function Laravel\Prompts\info;
+use function Laravel\Prompts\multiselect;
+use function Laravel\Prompts\note;
+
+class PullCommand extends Command
+{
+    protected $signature =
+        'pull {recipe?*}' .
+        '{--no-process : prevent processes from executing}';
+
+    protected $description = 'Pull a recipe from mise.dev';
+
+    public function handle(): void
+    {
+        if ($this->option('no-process')) {
+            info('Dry run enabled');
+            Process::fake();
+        }
+
+        $remoteRecipes = $this->selectRemoteRecipes();
+
+        foreach ($remoteRecipes as $recipe) {
+            $this->install($recipe);
+        }
+
+        if (confirm('Do you want to run the recipe now?')) {
+            $this->call('apply', [
+                'recipe' => $remoteRecipes,
+                '--no-process' => $this->option('no-process'),
+            ]);
+        }
+    }
+
+    protected function selectRemoteRecipes(): array
+    {
+        $selectedRemoteRecipes = $this->argument('recipe');
+
+        if (empty($selectedRemoteRecipes)) {
+            return multiselect(
+                label: 'Which recipe(s) should I pull?',
+                options: app(MiseService::class)->allForSelect(),
+            );
+        }
+
+        if (count($missingRecipes = array_diff($selectedRemoteRecipes, app(MiseService::class)->keys())) > 0) {
+            error('The following keys were not found and will be skipped');
+            note(collect($missingRecipes)->map(fn ($key) => "  {$key}")->implode("\n"));
+        }
+
+        return app(MiseService::class)->keys()->filter(
+            fn (string $key) => in_array($key, $selectedRemoteRecipes)
+        )->toArray();
+    }
+
+    private function install($key)
+    {
+        info('Installing recipe: ' . $key);
```

Comment:
I'm never totally sure about best practices here, but is there any value in us shipping a hash of the file contents? I know people do that, but I'm trying to figure out if there's any way that helps us in this context.